src/utils/interpolateTemplate.ts

```diff
@@ -0,0 +1,40 @@
+interface MessagesInterface {
```

Comment:
I'd like to see some tests for the `interpolateTemplate` method, given that you've got a fair amount of logic in there, and everything else in the tests depend on this working properly.

Alternatively, you could put all the tests back to working how they did before (i.e. just assert on all the strings as before), and then you've got validation that we've not regressed.

Reply:
Fair point. I've undone the changes to the spec files.